apps/web/src/hooks/useVariablesManager.ts

```diff
@@ -1,40 +1,47 @@
-import { useCallback, useLayoutEffect, useState } from 'react';
+import { useLayoutEffect, useState } from 'react';
 import { useFieldArray, useFormContext } from 'react-hook-form';
 import { parse } from '@handlebars/parser';
 import isEqual from 'lodash.isequal';
 import { getTemplateVariables } from '@novu/shared';
+
 import { IForm, ITemplates } from '../pages/templates/components/formTypes';
+import { useStepFormPath } from '../pages/templates/hooks/useStepFormPath';
+import { useStepIndex } from '../pages/templates/hooks/useStepIndex';
 
-export const useVariablesManager = (index: number, contents: string[]) => {
-  const { watch, control, getValues } = useFormContext<IForm>();
-  const variablesArray = useFieldArray({ control, name: `steps.${index}.template.variables` });
-  const variableArray = watch(`steps.${index}.template.variables`);
+const getTextContent = ({ templateToParse, fields }: { templateToParse?: ITemplates; fields: string[] }): string => {
+  return fields
+    .map((con) => con.split('.').reduce((a, b) => a && a[b], templateToParse ?? {}))
+    .map((con) => (Array.isArray(con) ? con.map((innerCon) => innerCon.content).join(' ') : con))
+    .join(' ');
+};
 
-  const getTextContent = useCallback(
-    ({ templateToParse, fields }: { templateToParse?: ITemplates; fields: string[] }): string => {
-      return fields
-        .map((con) => con.split('.').reduce((a, b) => a && a[b], templateToParse ?? {}))
-        .map((con) => (Array.isArray(con) ? con.map((innerCon) => innerCon.content).join(' ') : con))
-        .join(' ');
-    },
-    []
-  );
+export const useVariablesManager = (contents: string[]) => {
+  const { stepIndex, variantIndex } = useStepIndex();
+  const { watch, control, getValues } = useFormContext<IForm>();
+  const stepFormPath = useStepFormPath();
+  const variablesArray = useFieldArray({ control, name: `${stepFormPath}.template.variables` });
+  const variableArray = watch(`${stepFormPath}.template.variables`);
```

Comment:
use the "dynamic path" to the form variables array... dynamic means that depending on the URL it will be a path to step or to variant